compiler/rustc_resolve/src/macros.rs

```diff
@@ -544,11 +544,19 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
         if let Some((article, expected)) = unexpected_res {
             let path_str = pprust::path_to_string(path);
             let msg = format!("expected {}, found {} `{}`", expected, res.descr(), path_str);
-            self.tcx
-                .sess
-                .struct_span_err(path.span, &msg)
-                .span_label(path.span, format!("not {} {}", article, expected))
-                .emit();
+            let mut err = self.tcx.sess.struct_span_err(path.span, &msg);
+
+            err.span_label(path.span, format!("not {} {}", article, expected));
+
+            if !path.span.from_expansion() {
+                // Suggest moving the macro out of the derive() as the macro isn't Derive
+                if kind == MacroKind::Derive && ext.macro_kind() != MacroKind::Derive {
+                    err.span_help(path.span, "Remove from the surrounding `derive()`");
```

Comment:
```suggestion
                    err.span_help(path.span, "remove from the surrounding `derive()`");
```
diagnostic messages always start with lowercase